File: cs/XDust/XDust/XDustTextNode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace XDust
{
    public class XDustTextNode : XDustNode
    {
        public XDustTextNode(String value)
            : base()
        {
            this.Buffer = new StringBuilder();
            if (null != value)
            {
                this.Buffer.Append(value);
            }
        }

        private StringBuilder Buffer { get; set; }
        private String Value { get; set; }

        public void Write(Object value)
        {
            this.Buffer.Append(value);
        }

        public void Close()
        {
            if (null != this.Buffer)
            {
                this.Value = this.Buffer.ToString();
                this.Buffer = null;
            }
        }

        public override string Render(XDust dust, RenderChain chain, Context context, object model)
        {
            this.Close();
            return this.Value;
        }

        public override string ToString()
        {
            this.Close();
            return this.Value;
        }
    }
}

File: java/src/xdust/src/com/heydanno/xdust/XDustTextNode.java
package com.heydanno.xdust;

public class XDustTextNode extends XDustNode {
	
	public XDustTextNode(String value) {
		this.buffer = new StringBuilder();
		if (null != value) {
			this.buffer.append(value);
		}
	}
	
	private StringBuilder buffer;
	private String value;
	
	public void write(Object value) {
		this.buffer.append(value);
	}
	
	public void close() {
		if (null != this.buffer) {
			this.value = this.buffer.toString();
			this.buffer = null;
		}
	}
	
	@Override
	public String render(XDust dust, RenderChain chain, Context context,
			Object model) {
		return this.toString();
	}
	
	@Override
	public String toString() {
		this.close();
		return this.value;
	}

}
